src/app/content/highlights/components/EditCard.tsx

```diff
@@ -109,6 +120,20 @@ const EditCard = React.forwardRef<HTMLElement, Props>((
     onCancel();
   };
 
+  const updateUnsavedHighlightStatus = (newValue: string) => {
+    if (!data) {
+      return;
+    }
+
+    if (data.annotation !== newValue && !hasUnsavedHighlight) {
```

Comment:
you're gonna hit that undefined/empty string situation here too. might be nice to extract `const currentValue = data.annotation || ''` at the top of the logic